fuel-vm/src/tests/blockchain.rs

```diff
@@ -552,29 +560,37 @@ fn ldc_reason_helper(
 
 #[test]
 fn ldc_ssp_not_sp() {
-    // test ssp != sp for LDC
-    let load_contract = vec![
-        op::cfei(0x1), // sp += 1
-        op::ldc(RegId::ZERO, RegId::ZERO, RegId::ZERO), /* Load first two words from
-                        * the contract */
-    ];
+    let (load_contract, _) = script_with_data_offset!(
+        data_offset,
+        vec![
+            op::movi(0x10, data_offset as Immediate18),
+            op::cfei(0x1), // sp += 1
+            op::ldc(0x10, RegId::ZERO, RegId::ZERO), /* Load first two words from *
+                            * the contract */
+        ],
+        TxParameters::DEFAULT.tx_offset()
+    );
 
-    ldc_reason_helper(load_contract, ExpectedUnallocatedStack, false);
+    ldc_reason_helper(load_contract, ExpectedUnallocatedStack);
 }
 
 #[test]
 fn ldc_mem_offset_above_reg_hp() {
     // Then deploy another contract that attempts to read the first one
     let reg_a = 0x20;
 
-    // test memory offset above reg_hp value
-    let load_contract = vec![
-        op::move_(reg_a, RegId::HP), // r[a] := $hp
-        op::ldc(RegId::ZERO, RegId::ZERO, reg_a), /* Load first two words from the
-                                      * contract */
-    ];
+    let (load_contract, _) = script_with_data_offset!(
+        data_offset,
+        vec![
+            op::movi(0x10, data_offset as Immediate18),
+            op::move_(reg_a, RegId::HP), // r[a] := $hp
+            op::ldc(0x10, RegId::ZERO, reg_a), /* Load first two words from the
+                                          * contract */
```

Comment:
The comment is incorrect here. This loads first `$hp` bytes from the contract. The code should likely read like this:

```suggestion
            op::ldc(0x10, RegId::ZERO, RegId::HP),
```

Reply:
Thanks. I think a lot of these are wrong, but I don't really understand them enough to fix them. I'm just copying what was already there.

Might be worth fixing all of them, maybe in another PR.